### PR TITLE
#16379 Tidy up predefined icon_type support in Addon / devhub

### LINKED ISSUE #16379: Tidy up predefined icon_type support in Addon / devhub
Now #16114 has turned all predefined icons into custom icons, there are no add-ons remaining that have a predefined icons selected - only custom icons or the default puzzle piece.  

So we can remove all code that relies on `icon_type` starting with `icon/`; the icons themselves in /static/; and the js/template support for selecting an icon from a list.  (some of this could be spun out into separate issues)

### PR DESCRIPTION
**Pull request by Mitrajit**
Fixes #16379 

@eviljeff 
Changed the files in src/olympia/addons and src/olympia/devhub that checks whether `icon_type` starts with `icon/`.
Removed `create_custom_icon_from_predefined` function from src/olympia/addons/tasks.py
Due to removal of `create_custom_icon_from_predefined` function import errors are found in
- src/olympia/addons/management/commands/process_addons.py
- src/olympia/addons/tests/test_commands.py
- src/olympia/lib/settings_base.py
Let me know how shall I solve these.

### CODE REVIEW

**Comment by eviljeff** [comment]
you have a number of test fails to look at

**Comment by Mitrajit** [comment]
@eviljeff 

> Due to removal of create_custom_icon_from_predefined function import errors are found in
> - src/olympia/addons/management/commands/process_addons.py
https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/addons/management/commands/process_addons.py#L7-L9
> - src/olympia/addons/tests/test_commands.py
https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/addons/tests/test_commands.py#L626
> - src/olympia/lib/settings_base.py
https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/lib/settings_base.py#L1118

Most problems are from these three files how/what shall I modify in these files?

The rest of the errors are from code quality due to some unused variables, imports and ...
https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/devhub/tests/test_views_edit.py#L990 https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/devhub/tests/test_views_edit.py#L827 https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/amo/tests/test_celery.py#L49 Here `task_name =  'olympia.addons.tasks.create_custom_icon_from_predefined'` is causing problem

**Comment by eviljeff** [comment]
> > Due to removal of create_custom_icon_from_predefined function import errors are found in
> > 
> > * src/olympia/addons/management/commands/process_addons.py
> >   https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/addons/management/commands/process_addons.py#L7-L9
> > * src/olympia/addons/tests/test_commands.py
> >   https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/addons/tests/test_commands.py#L626
> > * src/olympia/lib/settings_base.py
> >   https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/lib/settings_base.py#L1118
> 
> Most problems are from these three files how/what shall I modify in these files?

These are all because the underlying `create_custom_icon_from_predefined` task was removed.  So you can remove the dict entries in process_addons.py and settings_base.py for `create_custom_icon_from_predefined`, and the `TestCreateCustomIconFromPredefined` class.

> The rest of the errors are from code quality due to some unused variables, imports and ...
> 
> https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/devhub/tests/test_views_edit.py#L990

you can delete this

> https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/devhub/tests/test_views_edit.py#L827

and this too

> https://github.com/mozilla/addons-server/blob/47a930271d31ceaa2d7a0770391234bf71f26711/src/olympia/amo/tests/test_celery.py#L49
> Here `task_name = 'olympia.addons.tasks.create_custom_icon_from_predefined'` is causing problem

this will go away when the entry is deleted from settings_base.py

**Comment by Mitrajit** [comment]
@eviljeff All pytest tests works now but cannot understand why I am getting codestyle error
```
black --check src/ services/ tests/
would reformat /home/circleci/project/src/olympia/devhub/tests/test_views_edit.py
Oh no! 💥 💔 💥
1 file would be reformatted, 648 files would be left unchanged.
make: *** [Makefile-docker:190: lint] Error 1
```

**Comment by eviljeff** [comment]
@Mitrajit you can run `make format` to fix formatting.

File: src/olympia/devhub/forms.py [hunk 1]
```diff
@@ -317,7 +317,6 @@ class Meta:
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
```
Comment (eviljeff) [inline src/olympia/devhub/forms.py:319]: you can delete the `__init__` method now - it's not doing anything beyond the superclass.

Comment (Mitrajit) [reply src/olympia/devhub/forms.py:319]: sure thing

**Comment by eviljeff** [comment]
I missed something, you can delete all the files in https://github.com/mozilla/addons-server/tree/master/static/img/addon-icons - except the `default-` files - because they're no longer used by the code you removed.

**Comment by Mitrajit** [comment]
@eviljeff I have deleted those unused images.

**Review by eviljeff — APPROVED** [review]
great, thanks r+

**Comment by Mitrajit** [comment]
@eviljeff you're most welcome. Which issue shall I checkout next?

**Comment by eviljeff** [comment]
> @eviljeff you're most welcome. Which issue shall I checkout next?

you can look though the other contrib:welcome issues

**Comment by caitmuenster** [comment]
@Mitrajit We'd love to have you take a look at the contrib:welcome issues! Take a look at [this list](https://addons-pm.herokuapp.com/contrib/contrib-welcome/?dir=desc&sort=updatedAt) and let us know if anything looks interesting. :)

**Comment by Mitrajit** [comment]
@caitmuenster Sure, I will check them out and let you know.

**Pull request merged**